index.js

```diff
@@ -751,6 +827,32 @@
            keys.every(function(k) { return equals(self[k], other[k]); });
   }
 
+  function unionKeys(a, b) {
+    var keys = {};
+    for (var key in a) keys[key] = 0;
+    for (key in b) keys[key] = 0;
+    return Object.keys(keys);
+  }
+
+  //  Object$prototype$lte :: StrMap a ~> StrMap a -> Boolean
+  function Object$prototype$lte(other) {
+    var keys = unionKeys(this, other).sort();
+    for (var idx = 0; idx < keys.length; idx += 1) {
+      var key = keys[idx];
+      var a = this[key], b = other[key];
+
+      if (!equals(a, b)) {
+        var inThis = Object.prototype.propertyIsEnumerable.call(this, key);
+        var inOther = Object.prototype.propertyIsEnumerable.call(other, key);
+        return  inThis === false ? true :
+                inOther === false ? false :
+                lte(a, b);
+      }
+    }
+
+    return true;
+  }
+
```

Comment:
I like that they're much cleaner. The implementation of `Object$prototype$lte` changes the semantics though. Previously, `Z.lte({x: 0, y: 0, z: 0}, {y: 0, z: 1}) === false` as this was conceptually testing:
```js
Z.lte([0, 0, 0], [undefined, 0, 1])
```
See https://github.com/sanctuary-js/sanctuary-type-classes/pull/46#issuecomment-293663721